lib/trento/application/process_managers/deregistration_process_manager.ex

```diff
@@ -19,61 +32,108 @@ defmodule Trento.DeregistrationProcessManager do
 
   deftype do
     field :cluster_id, Ecto.UUID
+    embeds_many :application_instances, Instance
+    embeds_many :database_instances, Instance
   end
 
   alias Trento.DeregistrationProcessManager
 
   alias Trento.Domain.Events.{
+    ApplicationInstanceDeregistered,
+    ApplicationInstanceRegistered,
     ClusterRolledUp,
+    DatabaseInstanceDeregistered,
+    DatabaseInstanceRegistered,
     HostAddedToCluster,
     HostDeregistered,
     HostDeregistrationRequested,
     HostRegistered,
     HostRemovedFromCluster,
-    HostRolledUp
+    HostRolledUp,
+    SapSystemRolledUp
   }
 
   alias Trento.Domain.Commands.{
+    DeregisterApplicationInstance,
     DeregisterClusterHost,
+    DeregisterDatabaseInstance,
     DeregisterHost
   }
 
-  @doc """
-    The process manager is interested in HostRegistered which starts or joins an existing process
-    manager for the host identified by a host_id field.
-    The process manager is interested in HostDeregistered which stops a process manager for the host identified by host_id.
+  alias Trento.Domain.SapSystem
 
-    We consider also the host rollup case, starting a process manager when the host rolled up arrived, so we identify a registered host
-    without the HostRegistered event, because that event could be rolled up and then we have to consider also the rolled up host as registered.
+  @doc """
+    The Process Manager is started by the following events (provided the instance hasn't been started already):
+    - HostRegistered starts a Process Manager for a newly registered host.
```

Comment:
The new doc looks great.
So many `starts a Process Manager` look redundant.
What about:
```
- HostRegistered for a newly registered host/
- HostAddedToCluster when a Host gets added to a cluster, as this event may arrive prior to ...
- DatabaseInstanceRegistered when an instance gets...
```